### ISSUE TITLE
ygm::io::ndjson_parser duplicate read

### ISSUE THREAD

**Issue by KIwabuchi** [LLNL/ygm#90]
It looks that `ndjson_parser` reads one of the lines in a file twice when there are more than a certain number of MPI ranks.
Here is a minimum code that produces the issue.

```C++
#include <iostream>
#include <ygm/comm.hpp>
#include <ygm/io/ndjson_parser.hpp>

int main(int argc, char **argv) {

  ygm::comm comm(&argc, &argv);
  {
    ygm::io::ndjson_parser jsonp(comm, std::vector<std::string>{"/p/lustre1/llamag/OPTC_E_Final_Foraker_JSON/process_base.json"});
    std::size_t count = 0;
    auto process_base_json = [&count](auto &j) {
      ++count;
    };
    jsonp.for_all(process_base_json);
    comm.barrier();
    comm.cout0() << "Read lines: " << comm.all_reduce_sum(count) << std::endl;
  }

  return 0;
}
```

```bash
$ wc -l /p/lustre1/llamag/OPTC_E_Final_Foraker_JSON/process_base.json
5211292 /p/lustre1/llamag/OPTC_E_Final_Foraker_JSON/process_base.json

$ srun --tasks-per-node=4 -N 8 ./src/test_ndjson_parser
Read lines: 5211292

$ srun --tasks-per-node=8 -N 8 ./src/test_ndjson_parser
Read lines: 5211292

$ srun --tasks-per-node=16 -N 8 ./src/test_ndjson_parser
Read lines: 5211293

$ srun --tasks-per-node=16 -N 16 ./src/test_ndjson_parser
Read lines: 5211293
```

I'm using YGM's latest develop commit (4d7e98abeaa031198328112a3982d07ca8c6a011).

Could you look into that?

**Comment by steiltre** [comment]
This is related to the bug fixed in a previous [PR](https://github.com/LLNL/ygm/pull/64). The changes may have only fixed the issue in certain cases or may have introduced a new edge case that needs fixing.

I'll look into this and fix it in the next couple of days.

**Comment by KIwabuchi** [comment]
Just checking in. Did you find something? I also looked into the code, but it looks good to me🤔

**Comment by steiltre** [comment]
Not yet. Starting to look now, so hopefully I'll have a fix today.

**Comment by steiltre** [comment]
This is addressed in [PR #93](https://github.com/LLNL/ygm/pull/93), which will be merged in once CI jobs complete.

This fixes the issue on the example file provided. Let me know if this issue seems to still exist in other data after the PR. It's always possible that I'm just shifting around the places this bug shows up again...

**Comment by KIwabuchi** [comment]
It looks good to me! I'm closing this issue.
